src/dfmt/indentation.d

```diff
@@ -67,16 +67,21 @@ struct IndentStack
      */
     void push(IdType item) pure nothrow
     {
-        arr[index] = item;
-        index = index + 1 == arr.length ? index : index + 1;
+        arr[index++] = item;
+        if (index == arr.length)
+        {
+            //FIXME this is actually a bad thing to do
+            index--;
+        }
```

Comment:
I think you should just do ` arr[index]=item;  if ( index < arr.length ) ++index; `

Reply:
That's  less brittle, yes.
But I'd like it to stay brittle such that I have more arguments in favour for changing how this is done.

Reply:
oops this can be read by everyone ....

Reply:
It can.